app/javascript/components/analytics/widgets/Demographics.js

```diff
@@ -1,191 +1,249 @@
+  'More Than One Race',
+  'Unknown',
+];
+// const SEXUAL_ORIENTATIONS = ['Straight or Heterosexual', 'Lesbian, Gay, or Homosexual', 'Bisexual', 'Another', 'Choose not to disclose', 'Don’t know'];
 
 class Demographics extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { checked: false, viewTotal: this.props.viewTotal };
-    this.handleChange = this.handleChange.bind(this);
-    this.toggleBetweenActiveAndTotal = this.toggleBetweenActiveAndTotal.bind(this);
-    this.obtainValueFromMonitoreeCounts = this.obtainValueFromMonitoreeCounts.bind(this);
-    this.ERRORS = !Object.prototype.hasOwnProperty.call(this.props.stats, 'monitoree_counts');
-    this.ERRORSTRING = this.ERRORS ? 'Incorrect Object Schema' : null;
-    if (!this.ERRORS) {
-      this.ageData = this.obtainValueFromMonitoreeCounts(AGEGROUPS, 'Age Group', this.state.viewTotal);
-      this.sexData = this.obtainValueFromMonitoreeCounts(SEXES, 'Sex', this.state.viewTotal);
-    }
-  }
+    this.ageData = this.parseOutFields(AGEGROUPS, 'Age Group');
+    this.sexData = this.parseOutFields(SEXES, 'Sex');
+    this.ethnicityData = this.parseOutFields(ETHNICITIES, 'Ethnicity');
+    this.raceData = this.parseOutFields(RACES, 'Race');
+    // this.soData = this.parseOutFields(SEXUAL_ORIENTATIONS, 'Sexual Orientation');
+    this.hasFakeBirthdateData = false;
+    this.numberOfFakeBirthdates = 0;
 
-  componentDidUpdate(prevProps) {
-    if (this.props.viewTotal !== prevProps.viewTotal) {
-      this.toggleBetweenActiveAndTotal(this.props.viewTotal);
+    // Meaning we have monitorees with fake birthdates
+    if (_.last(_.last(this.ageData)) !== 0) {
+      this.hasFakeBirthdateData = true;
+      this.numberOfFakeBirthdates = _.last(_.last(this.ageData));
+      let indexOfGreaterThan80 = AGEGROUPS.findIndex(x => x === '>=80');
+      // This overly complex statement just adds every value from the `110+` fields to the `>=80` fields
+      this.ageData[Number(indexOfGreaterThan80)] = this.ageData[Number(indexOfGreaterThan80)].map(
+        (x, i) => this.ageData[Number(indexOfGreaterThan80)][Number(i)] + _.last(this.ageData)[Number(i)]
+      );
     }
+
+    // Map and translate all of the Tabular Data to the Chart Format
+    this.ageChartData = this.mapToChartFormat(_.initial(AGEGROUPS), this.ageData);
+    this.sexChartData = this.mapToChartFormat(SEXES, this.sexData);
+    this.ethnicityChartData = this.mapToChartFormat(ETHNICITIES, this.ethnicityData);
+    this.raceChartData = this.mapToChartFormat(RACES, this.raceData);
+    // this.soChartData = this.mapToChartFormat(SEXUAL_ORIENTATIONS, this.soData);
+    this.barGraphData = [
+      { title: 'Age (Years)', data: this.ageChartData },
+      { title: 'Sex', data: this.sexChartData },
+      { title: 'Race', data: this.ethnicityChartData },
+      { title: 'Ethnicity', data: this.raceChartData },
+      // { title: 'Sexual Orientation', data: this.soChartData },
```

Comment:
It seems like a lot of the sexual orientation analytic (generation _and_ display)  has been commented out -- is this intentional?